solr/core/src/java/org/apache/solr/update/SolrIndexWriter.java

```diff
@@ -135,8 +135,7 @@ private SolrIndexWriter(SolrCore core, String name, String path, Directory direc
     infoStream = getConfig().getInfoStream();
     this.directory = directory;
     numOpens.incrementAndGet();
-    metricManager = core.getCoreContainer().getMetricManager();
-    registryName = core.getCoreMetricManager().getRegistryName();
+    solrMetricsContext = core.getSolrMetricsContext().getChildContext(this);
```

Comment:
Similar situation to `SolrIndexSearcher`, the life-cycle here is different from SolrCore.